lib/OrdinaryDiffEqCore/src/integrators/integrator_interface.jl

```diff
@@ -3,7 +3,7 @@
 # Hence, we need to have two separate functions.
 
 function _change_t_via_interpolation!(integrator, t,
-        modify_save_endpoint::Type{Val{T}}) where {T}
+        modify_save_endpoint::Type{Val{T}}, reinitialization_method=nothing) where {T}
```

Comment:
```suggestion
        modify_save_endpoint::Type{Val{T}}, initialize_alg=nothing) where {T}
```

Reply:
either this or callback_initializealg? The choice here doesn't follow the naming conventions or the other names.

Reply:
How about `reinitalize_alg`? I'm not super happy with simply calling it `initialize_alg` because it doesn't answer the question of "why does something that occurs mid-solve need to do initialization?" While "reinitialize" doesn't capture that super clearly, I think it's more obvious then just `initialize_alg`.